sdk/client/client.ts

```diff
@@ -124,39 +117,98 @@ export class Client {
   api: Api
+
   // The path to the secrets file which may be used to store credentials
-  secretsFilePath: string
+  secretsFilePath: string = process.cwd() + '/secrets.json'
+  secretsFileContent: { api: {} }
+
   // The credential access details
   private tokenUrlDetails: [Source, string]
   private usernameDetails: [Source, string]
   private passwordDetails: [Source, string]
   private clientIdDetails: [Source, string]
   private clientSecretDetails: [Source, string]
+
   // The refresh limit in seconds before token expiry to trigger a refresh
-  refreshLimit: number = refreshLimit
+  refreshLimit: number = 3580
+
+  private loadSecretsFile(): object | null {
+
+    try
+    {
+      this.secretsFileContent = require( this.secretsFilePath );
+    }
+    catch( e )
+    {
+      // not necessarily a problem. even if the format of the source file is bad, this will reset the internal validation to an empty configuration
+
+      this.secretsFileContent = { api: {} };
+    }
+
+    return this.secretsFileContent;
+
+  }
 
   // Constructor method which takes the details on where to find the credentials
   constructor(
-    tokenUrlDetails: [Source, string],
-    usernameDetails: [Source, string],
-    passwordDetails: [Source, string],
-    clientIdDetails: [Source, string],
-    clientSecretDetails: [Source, string],
-    apiUrlDetails: [Source, string],
+      {
+        tokenUrlDetails,
+        usernameDetails,
+        passwordDetails,
+        clientIdDetails,
+        clientSecretDetails,
+        apiUrlDetails
+      }: {
+        tokenUrlDetails?: [Source, string],
+        usernameDetails?: [Source, string],
+        passwordDetails?: [Source, string],
+        clientIdDetails?: [Source, string],
+        clientSecretDetails?: [Source, string],
+        apiUrlDetails?: [Source, string],
+      } = {
+        tokenUrlDetails: [Source.Secrets, 'tokenUrl'],
+        usernameDetails: [Source.Secrets, 'username'],
+        passwordDetails: [Source.Secrets, 'password'],
+        clientIdDetails: [Source.Secrets, 'clientId'],
+        clientSecretDetails: [Source.Secrets, 'clientSecret'],
+        apiUrlDetails: [Source.Secrets, 'apiUrl']
+      }
     ) {
 
-    // Set the path to the secrets file
-    this.secretsFilePath = secretsPath
-
-    // Set the credential details
-    this.tokenUrlDetails = tokenUrlDetails
-    this.usernameDetails = usernameDetails
-    this.passwordDetails = passwordDetails
-    this.clientIdDetails = clientIdDetails
-    this.clientSecretDetails = clientSecretDetails
-
-    // Set the base path for the API
-    this.basePath = this.fetchCredentials(apiUrlDetails[0], apiUrlDetails[1])
+    // attempt to get disk config by default. if it doesn't exist it is still fine
+    this.loadSecretsFile();
+
+    // provide init + default to environment vars below
+    this.tokenUrlDetails = !!tokenUrlDetails ? tokenUrlDetails : (
+      // [Source.Environment, 'FBN_TOKEN_URL']
+      [Source.Secrets, 'tokenUrl']
+    );
```

Comment:
If the defaults are already specified in the constructor above in what situation would  !!tokenUrlDetails resolve to False? In that situation what is the value of providing the same details "[Source.Secrets, 'tokenUrl']" as the default?